src/test/dbwrapper_tests.cpp

```diff
@@ -42,6 +42,95 @@ BOOST_AUTO_TEST_CASE(dbwrapper)
     }
 }
 
+BOOST_AUTO_TEST_CASE(dbwrapper_basic_data)
+{
+    // Perform tests both obfuscated and non-obfuscated.
+    for (bool obfuscate : {false, true}) {
+        fs::path ph = SetDataDir(std::string("dbwrapper_1").append(obfuscate ? "_true" : "_false"));
+        //CDBWrapper dbw(ph, (1 << 20), true, false, obfuscate);
+        CDBWrapper dbw(ph, (1 << 20), false, true, obfuscate);
+
+        uint256 res;
+        uint32_t res_uint_32;
+        bool res_bool;
+
+        // Ensure that we're doing real obfuscation when obfuscate=true
+        BOOST_CHECK(obfuscate != is_null_key(dbwrapper_private::GetObfuscateKey(dbw)));
+ 
+        //Simulate block raw data - "b + block hash"
+        std::string block_tag = "b";
+        uint256 block_hash = InsecureRand256();
+        std::string key_block = block_tag + block_hash.ToString();
```

Comment:
Inlining these constructions is a bit more clear IMO, because the var names are redundant with the comments:
```c++
//Simulate block raw data - "b + block hash"
std::string key_block = "b" + InsecureRand256().ToString();

//Simulate file raw data - "f + file_number"
std::string key_file = strprintf("f%04x", InsecureRand32());
```

Reply:
/Simulate block raw data - "b + block hash"
std::string key_block = "b" + InsecureRand256().ToString();

//Simulate file raw data - "f + file_number"
std::string key_file = strprintf("f%04x", InsecureRand32());